expo/scheduler/functions.py

```diff
@@ -436,3 +437,121 @@ def get_roles_from_scheduler(workeritems, conference):
             roles += [allocation.resource.as_subtype.role]
 
     return list(set(roles))
+
+
+def calendar_export(conference=None,
+                    cal_format='gbook',
+                    event_types='All',
+                    day=None,
+                    ):
+    '''
+    View to export calendars, formatted in either iCalendar format, or as
+    a csv file.  Used to allow importing of the calendar information into
+    other applications, such as Guidebook, or Calendar applications.
+
+    conference  - Slug for the conference we want to export data for
+    cal_format  - Which format to use for the exported data, can be 'gbook'
+                  for Guidebook csv, 'ical' for iCal calendar programs (ics)
+    event_types - Which event types to include, usually will be 'All' to
+                  generate a complete calendar, but could also be 'Class',
+                  'Show', etc, or a list or tuple of event_types
+    day         - Which day of the conference to get calendar info for,
+                  None or 'All' gets info for entire conference
+    '''
+    #  TODO: Add ability to filter on a users schedule for things like
+    #  volunteer shifts.
+
+    from gbe.functions import get_conference_by_slug
+    import gbe.models as conf
+    from expo.settings import (
+        DATETIME_FORMAT,
+        TIME_FORMAT,
+        SITE_URL,
+        )
+    import csv
+
+    if conference is None:
+        conference = conf.Conference.current_conf()
+    else:
+        conference = get_conference_by_slug(conference)
+
+    if day == 'All':
+        day = None
+    cal_times = cal_times_for_conf(conference, day)
+
+    local_domain = SITE_URL.split('://')[1].replace('www.', '')
+    if event_types == 'All':
+        event_types = ['Show',
+                       'Class',
+                       'Special Event',
+                       'Master Class',
+                       'Drop-In Class']
+    elif event_types == 'Show':
+        event_types == ['Show',
+                        'Special Event',
+                        'Master Class',
+                        'Drop-In Class',
+                        ]
+    elif not type(event_types).isinstance([]) or \
+            type(event_types).isinstance(()):
+        event_types = [event_types]
+    events = []
+    for event_type in event_types:
+        events = events + event_info(confitem_type=event_type,
+                                     cal_times=cal_times,
+                                     conference=conference)
+    if cal_format == 'gbook':
+        return_file = '"Session Title","Date","Time Start","Time End",' + \
+                      '"Room/Location","Schedule Track (Optional)",' + \
+                      '"Description (Optional)",""\r\n'
```

Comment:
We need all the headers they give...

Session Title	
Date	
Time Start	
Time End	
Room/Location	
Schedule Track (Optional)	
Description (Optional)	
**Allow Checkin (Optional)	
Checkin Begin (Optional)	
Limit Spaces? (Optional)	
Allow Waitlist (Optional)**

The optional ones can have no data at all, but they need to have headers.